std/uri.d

```diff
@@ -372,6 +372,47 @@ string encodeComponent(Char)(in Char[] uriComponent) if (isSomeChar!Char)
     return URI_Encode(s, URI_Alpha | URI_Digit | URI_Mark);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *
+ * Params:
+ *      values = an associative array containing the values to be encoded.
+ *
+ * Returns:
+ *      A string encoded using www-form-urlencoding.
+ */
+package string urlEncode(int reserve = -1)(in string[string] values)
```

Comment:
@burner Someone suggested I added a blank template param e.g:

```d
package string urlEncode()(in string[string] values) ...
```
This was before adding the reserve specialisation.

Should I just make the function concrete instead?